platforms/beepy/src/i2c_puppet.rs

```diff
@@ -0,0 +1,632 @@
+    /// is off, 255 is maximum brightness.
+    pub async fn backlight(&mut self) -> Result<u8, Error> {
+        if let Response::Backlight(brightness) = self.request(Request::GetBacklight).await? {
+            Ok(brightness)
+        } else {
+            unreachable!("service responded with wrong response variant!")
+        }
+    }
+
+    async fn request(&mut self, msg: Request) -> Result<Response, Error> {
+        self.handle
+            .request_oneshot(msg, &self.reply)
+            .await
+            .map_err(|error| {
+                tracing::warn!(?error, "failed to send request to i2c_puppet service");
+                Error::SendRequest(error)
+            })
+            .and_then(|resp| resp.body)
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Server Definition
+////////////////////////////////////////////////////////////////////////////////
+
+/// Server implementation for the [`I2cPuppetService`].
+pub struct I2cPuppetServer {
+    settings: I2cPuppetSettings,
+    rx: KConsumer<registry::Message<I2cPuppetService>>,
+    i2c: I2cClient,
+    subscriptions: FixedVec<KProducer<(KeyStatus, KeyRaw)>>,
+}
+
+#[derive(Debug)]
+pub enum RegistrationError {
+    Registry(registry::RegistrationError),
+    NoI2cPuppet(I2cError),
+}
+
+// https://github.com/solderparty/i2c_puppet#protocol
+const ADDR: u8 = 0x1f;
+
+//// i2c_puppet I2C registers
+mod reg {
+    /// To write with a register, we must OR the register number with this mask:
+    /// <https://github.com/solderparty/i2c_puppet#protocol>
+    pub(super) const WRITE: u8 = 0x80;
+
+    // RGB LED configuration registers:
+    // https://beepy.sqfmi.com/docs/firmware/rgb-led#set-rgb-color
+
+    /// Controls whether the RGB LED is on or off.
+    pub(super) const LED_ON: u8 = 0x20;
+
+    /// 8-bit RGB LED red value.
+    pub(super) const LED_R: u8 = 0x21;
+    /// 8-bit RGB LED green value.
+    pub(super) const LED_G: u8 = 0x22;
+    /// 8-bit RGB LED blue value.
+    pub(super) const LED_B: u8 = 0x23;
+}
+
+impl I2cPuppetServer {
+    #[instrument(level = Level::DEBUG, skip(kernel))]
+    pub async fn register(
+        kernel: &'static Kernel,
+        settings: I2cPuppetSettings,
+    ) -> Result<(), RegistrationError> {
+        let (tx, rx) = KChannel::new_async(settings.channel_capacity).await.split();
+        let mut i2c = I2cClient::from_registry(kernel).await;
+        let subscriptions = FixedVec::new(settings.max_subscriptions).await;
+        {
+            // first, make sure we can get the version...
+            let mut kbd = AsyncBbq10Kbd::new(&mut i2c);
+            let Version { major, minor } = kbd
+                .get_version()
+                .await
+                .map_err(RegistrationError::NoI2cPuppet)?;
+            tracing::info!("i2c_puppet firmware version: v{major}.{minor}");
+            // // and then, reset i2c_puppet
```

Comment:
More dead code

Reply:
And if we *don't* remove this, might want to check what the i2c_puppet firmware does on reset: technically it switches the 5v power that normally powers the mango pi (when we actually start using battery power).

We want to make sure we don't reset IT and then it resets US.

Reply:
okay yeah we don't want that